feat: create new snapcraft-pack action (#45)
canonical/craft-actions: README.md

- [Charmcraft](https://juju.is/)
- [Snapcraft](https://snapcraft.io/)
- [Rockcraft](https://documentation.ubuntu.com/rockcraft/stable)

The following sections will break down the different actions available in this
repository and how to use them.

## snapcraft/setup

Set up Snapcraft on the runner.

### Usage

```yaml
steps:
  - uses: canonical/craft-actions/snapcraft/setup@main
  - run: |
    snapcraft --version
```

This will install and configure LXD and Snapcraft.

#### Inputs

##### `channel`

Select a channel for the Snapcraft snap to be installed from. Defaults to `latest/stable`.

##### `revision`

Select a revision for the Snapcraft snap to be installed from. Overrides the `channel` option.

##### `lxd-channel`

Select a channel for the LXD snap to be installed from. Defaults to the current recommended channel for Snapcraft.

---

## snapcraft/pack

Set up Snapcraft and pack a snap.

### Usage

```yaml
steps:
  - uses: actions/checkout@v6
  - uses: canonical/craft-actions/snapcraft/pack@main
```

> This will install and configure LXD and Snapcraft, then pack a snap in the given directory.

Reviewer: ```suggestion
This will install and configure LXD and Snapcraft before packing the snap in the given directory.
```
Regarding 'a snap' versus 'the snap': is there a case where there can be multiple snaps in the directory the user provides?

Author: In theory, a repository absolutely could have multiple `snapcraft.yaml` files in different locations. It'd be a weird thing to do, and wouldn't work out of the box, but totally possible given the context here is "if you have your file in a weird spot", and it'd have to be in a weird spot to support multiple `snapcraft.yaml`s.

Reviewer: To be more specific, I'm asking about the path the user provides for packing. Can't there only be a single snap project in that directory?

Author: Oh, I see what you mean now with relation to the wording - yeah, it should only be possible to have one snap in a directory.

#### Inputs

##### `path`

The location in your repository to run Snapcraft when packing. Defaults to the repository root.

In the simplest cases, `snapcraft.yaml` is often stored in the `snap/` directory at the repository root.

In repositories with more complex file hierarchies, the path would be elsewhere. For example, if your project
has multiple build tool manifests stored in a `dist/` directory, and `snapcraft.yaml` is stored in `dist/snap/`,
you would want to run Snapcraft at `dist/`, rather than the repository root.

```yaml
steps:
  - uses: actions/checkout@v6
  - uses: canonical/craft-actions/snapcraft/pack@main
    with:
      path: dist/
```

##### `verbosity`

> Used to set the build verbosity level for Snapcraft. Must be one of "quiet", "brief", "verbose", "debug", or "trace" (the default).

Reviewer: ```suggestion
The build verbosity level for Snapcraft. Must be one of: "quiet", "brief", "verbose", "debug", or "trace" (the default).
```
While I prefer what I've suggested here, it'd be more consistent with the other actions to start the sentence with "Sets the build verbosity..."

I'm not too picky about this, and I also have no qualms with doing a quick pass on the other descriptions if need be.

```yaml
steps:
  - uses: actions/checkout@v3
  - uses: canonical/craft-actions/charmcraft/pack@main
    with:
      verbosity: debug
```

##### `channel`

Select a channel for the Snapcraft snap to be installed from. Defaults to `latest/stable`.

##### `revision`

> Select a revision for the Snapcraft snap to be installed from. Overrides the `channel` option.

Author: Almost correct - it would pull the specified revision regardless of order (but I think that's what you meant to type)
